src/components/Search/SearchComponent.tsx

```diff
@@ -87,33 +71,29 @@ function SearchComponent({
       },
     },
   });
-  const cancelBtn = {
-    main: style({
-      $nest: {
-        '&.MuiButtonBase-root': {
-          ...btnStyles.Secondary,
-        },
-      },
-    }),
-  };
 
-  const dispatch = useAppDispatch();
+  const { search } = useAppSelector((state) => state);
+  const baseRepository = search.base.repository;
+  const newRepository = search.new.repository;
   const checkedRevisionsList = useAppSelector(
     (state) => state.search[searchType].checkedRevisions,
   );
-  const searchState = useAppSelector((state) => state.search[searchType]);
   const selectedRevisions = useAppSelector(
     (state) => state.selectedRevisions.revisions,
   );
-  const { editSelectedRevisions, updateSelectedRevisions } =
-    useSelectedRevisions();
 
+  const searchState = useAppSelector((state) => state.search[searchType]);
   const { searchResults } = searchState;
   const [focused, setFocused] = useState(false);
-  const [dropDownInputVisible, setDropDownInputVisible] = useState(true);
-  const [editBtnVisible, toggleEditBtnVisible] = useState(true);
+  const location = useLocation();
+  const view = location.pathname == '/' ? searchView : compareView;
   const matchesQuery = useMediaQuery('(max-width:768px)');
   const containerRef = createRef<HTMLDivElement>();
+  const editButtonRef = createRef<HTMLButtonElement>();
+  const selectedRevisionsRef = createRef<HTMLDivElement>();
+  const isWarning =
+    (baseRepository === 'try' && newRepository !== 'try') ||
+    (baseRepository !== 'try' && newRepository === 'try');
```

Comment:
It's not super clear to me that this is the right place, I wonder if `CompareWithBase` was better. Indeed `SearchComponent` is either `base` or `new`, so doing this check here doesn't feel right to me.